### PR TITLE
add possibility to filter data in BSTable on server side for DSS dataset

### PR DESCRIPTION
**Pull request by helaot**

### CODE REVIEW

File: commons/_shared/bs-lib-components/ui/dist/types/components/table/BsDSSTableFunctional.vue.d.ts [hunk 1]
```diff
@@ -23,12 +24,13 @@ declare const _sfc_main: import("vue").DefineComponent<{
...
     setFetchingChunk(fetchingChunk: boolean): void;
     fetchDSSData(datasetName: string, chunksize?: number | undefined, chunkIndex?: number | undefined): Promise<Record<string, any>[] | undefined>;
+    fetchFilteredDSSDataset(datasetName: string, chunksize: number, chunk_index: number, filters?: Record<string, any[]> | undefined): Promise<Record<string, any>[] | undefined>;
     fetchDSSColumns(datasetName: string): Promise<{
         columns: BsTableCol[];
         columnsCount: number;
     }>;
     updateColumns(datasetName: string): void;
-    updateRows(datasetName: string, chunksize?: number | undefined, chunkIndex?: number | undefined): void;
+    updateRows(datasetName: string, chunksize: number, chunk_index: number, filters?: Record<string, any[]> | undefined): void;
```
Comment (levkz) [inline commons/_shared/bs-lib-components/ui/dist/types/components/table/BsDSSTableFunctional.vue.d.ts:33]: filters should have a type of `Records<string, string[]>`

Comment (levkz) [inline commons/_shared/bs-lib-components/ui/dist/types/components/table/BsDSSTableFunctional.vue.d.ts:33]: filters should have a type of `Records<string, string[]>`

**Review by levkz — APPROVED** [review]
LGTM, good job!

As a future improvement you could let the user define filters as `Record<string, string | string[]>`, so that the syntax: `{colName: colValue}` would be automatically transformed to `{colName: [colValue]}`.

File: commons/_shared/bs-lib-components/ui/dist/types/components/table/BsDSSTableFunctional.vue.d.ts (hunk 1 above)
Comment (helaot) [reply commons/_shared/bs-lib-components/ui/dist/types/components/table/BsDSSTableFunctional.vue.d.ts:33]: cant we have numbers for exp
?

**Pull request merged**